katdal/visdatav4.py

```diff
@@ -343,11 +334,26 @@ def __init__(self, source, ref_ant='', time_offset=0.0, applycal='',
         available_products = [product for product in CAL_PRODUCTS
                               if has_cal_product(self.sensor, attrs, product)]
         self._applycal = _selection_to_list(applycal, all=available_products)
+        if not self.source.data or not self._applycal:
+            self.correction = None
+            self._corrected = self.source.data
+        else:
+            self._correction = calc_correction(self.source.data.vis.chunks, self.sensor,
+                                               self.subarrays[self.subarray].corr_products,
+                                               self._applycal)
+            corrected_vis = self._make_corrected(apply_vis_correction, self.source.data.vis)
+            corrected_flags = self._make_corrected(apply_flags_correction, self.source.data.flags)
+            corrected_weights = self._make_corrected(apply_weights_correction, self.source.data.weights)
+            self._corrected = VisFlagsWeights(corrected_vis, corrected_flags, corrected_weights,
+                                              name='corrected')
```

Comment:
I'm not fussy, and I don't have a clear picture of what else you're stuffing into `name`. Pick something and I will make it so. Another option would be to incorporate `self.source.data.name` into the name so that you know where it's coming from.

Reply:
How about starting with `self.source.data.name`? If it contains 'l0', replace it with 'l1', else append '_corrected'.

Reply:
Done, although slightly differently because I didn't have your exact recipe open when I did the change. Feel free to tweak it on the branch.

Reply:
Done.